spec/topological_inventory/host_inventory_sync/host_inventory_sync_spec.rb

```diff
@@ -1,7 +1,53 @@
 require "topological_inventory/host_inventory_sync"
 
 RSpec.describe TopologicalInventory::HostInventorySync do
-  let(:parser) { described_class.new(openshift_host: "localhost") }
+  context "#topological_inventory_api (private)" do
+
+    it "returns the initial url if provided" do
+      expect(described_class.new("http://example.com/api/", "", "", 9092).send(:topological_inventory_api)).to eq("http://example.com/api/")
+    end
+
+    context "with service env vars set" do
+      let(:url) { described_class.new(nil, "", "", 9092).send(:topological_inventory_api) }
+
+      around do |e|
+        ENV["TOPOLOGICAL_INVENTORY_API_SERVICE_HOST"] = "example.com"
+        ENV["TOPOLOGICAL_INVENTORY_API_SERVICE_PORT"] = "8080"
```

Comment:
minor, but this can be done more simply with `stub_const`, which auto-cleans-up...

```ruby
stub_const(:ENV, {"TOPOLOGICAL_INVENTORY_API_SERVICE_HOST" => "example.com", "TOPOLOGICAL_INVENTORY_API_SERVICE_PORT" => "8080"})
```

Reply:
Alternately.

```ruby
allow(ENV).to receive(:[]).with("TOPOLOGICAL_INVENTORY_API_SERVICE_HOST").and_return("example.com")
```